src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import com.google.common.collect.Sets;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.*;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException.BadArgumentValue;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.GenotypeAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.InfoFieldAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.VariantAnnotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.ActiveRegionBased;
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.Standard;
+
+/**
+ * The class responsible for computing annotations for variants.
+ */
+public final class VariantAnnotatorEngine {
```

Comment:
not ported yet - it's used in `VariantAnnotator` which is not part of this pull req